crates/context_aware_config/migrations/2024-05-06-133756_custom_types/up.sql

```diff
@@ -0,0 +1,40 @@
+-- Your SQL goes here
+CREATE TABLE jsonschema_types (
+    type_name TEXT PRIMARY KEY,
+    type_schema JSON NOT NULL,
+    created_by TEXT NOT NULL,
+    created_at TIMESTAMP NOT NULL,
+    last_modified TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+CREATE INDEX IF NOT EXISTS json_schema_types_index ON ONLY public.jsonschema_types(type_name, created_at);
+INSERT INTO jsonschema_types(type_name, type_schema, created_by, created_at)
+VALUES (
+        'Number',
+        '{"type": "number"}',
+        'user@superposition.io',
+        NOW()
+    ),
+    (
+        'Decimal',
+        '{"type": "number"}',
+        'user@superposition.io',
+        NOW()
+    ),
+    (
+        'Boolean',
+        '{"type": "boolean"}',
+        'user@superposition.io',
+        NOW()
+    ),
+    (
+        'String (Enum)',
```

Comment:
I think we should restrict the type name to have only permitted characters.  Something like function name in Cloudfront:

https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_CreateFunction.html#API_CreateFunction_RequestSyntax

Reply:
@knutties I will add validation for the type name here. I want to allow spaces since it doesn't seem to break anything, if someone adds too many spaces when trying to create a type they can quickly delete it too

Reply:
@Datron - it will start breaking when we use function/types names as files (unless people take care).  For e.g. people might maintain some of the types/functions in their git repository to control edits to the functions (just like how we have a repo for AWS CFFs) - think infra as code.  In such cases these types might exist as files with the name of the type being the file name.  In such cases, scripts written have to be careful and handle spaces in file names correctly.

Reply:
@knutties this makes sense, I hadn't considered this. I will allow [a-zA-Z_]+ as a pattern?

Reply:
AWS uses Pattern: `^[a-zA-Z0-9-_]{1,64}$` for functions - which sounds like a good one to use.  We can use the same for functions as well.

Reply:
Should we allow numbers @knutties? I think text is enough

Reply:
Don't want to be too restrictive - particularly when people are playing around - numbers are sometimes handy to disambiguate stuff.